### PR TITLE
JAMES-2610 MailetPipelineLogging should always log a non empty mailet…

### PR DESCRIPTION
**Pull request by aduprat**
… info

### CODE REVIEW

File: mailet/base/src/test/java/org/apache/mailet/base/MailetPipelineLoggingTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * under the License.
+ */
+
+package org.apache.mailet.base;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.mailet.Mail;
+import org.junit.Before;
+import org.junit.Test;
```
Comment (chibenwa) [inline mailet/base/src/test/java/org/apache/mailet/base/MailetPipelineLoggingTest.java]: junit 5 ?

Comment (aduprat) [reply mailet/base/src/test/java/org/apache/mailet/base/MailetPipelineLoggingTest.java]: Not in this project yet & no added value for such tests, IMO.

File: mailet/base/src/main/java/org/apache/mailet/base/MailetPipelineLogging.java [hunk 2]
```diff
@@ -19,19 +19,31 @@
...
 
+
     public static void logEndOfMailetProcess(Mailet mailet, Mail mail) {
-        LOGGER.debug("Mail: {} End of mailet: {}", mail.getState(), mailet.getMailetInfo());
+        LOGGER.debug("Mail: {} End of mailet: {}", mail.getState(), getMailetInfo(mailet));
+    }
+
+    @VisibleForTesting
+    static String getMailetInfo(Mailet mailet) {
+        return Optional.ofNullable(Strings.emptyToNull(mailet.getMailetInfo()))
```
Comment (mbaechler) [inline mailet/base/src/main/java/org/apache/mailet/base/MailetPipelineLogging.java]: What about fixing the root of the problem by not allowing null or empty value ?

Comment (aduprat) [reply mailet/base/src/main/java/org/apache/mailet/base/MailetPipelineLogging.java]: This means an API change, then we will not be able to backport this fix.

**Comment by trantienduchn** [comment]
```
[3caaf70c6ca1cd4229e65a8c6ed16671caffbfe1] [ERROR] Errors: 
[3caaf70c6ca1cd4229e65a8c6ed16671caffbfe1] [ERROR]   SwiftTempAuthObjectStorageBlobsDAOBuilderTest.builtBlobsDAOCanStoreAndRetrieve:94 » Timeout
[3caaf70c6ca1cd4229e65a8c6ed16671caffbfe1] [INFO] 
[3caaf70c6ca1cd4229e65a8c6ed16671caffbfe1] [ERROR] Tests run: 124, Failures: 0, Errors: 1, Skipped: 0
```
test this please

**Pull request merged**